source/contracts/architecture.md

```diff
@@ -0,0 +1,108 @@
+
+The nonce prevents replay attacks. Each transaction should have a different nonce and once a transaction with a specific nonce has been executed it should not be possible to execute this transaction again. The concrete replay protection mechanism depends on the version of the Gnosis Safe and will be explained later.
+
+## Contract Creations
+As the creation of new contracts is a very gas consuming operation, Safe contracts use a proxy pattern where only one master copy of a contract is deployed once and all its copies are deployed as minimal proxy contracts pointing to the master copy contract. This pattern also allows to update the contract functionality later on by updating the address of the master copy in the proxy contract. 
+
+As contract constructors can only be executed once at the time the master copy is deployed, constructor logic has to be moved into an additional persistent setup function, which can be called to setup all copies of the master copy. This setup function has to be implemented in a way it can only be executed once. It is important to note that the master copy contract has to be persistent and there should be no possibility to execute a `selfdestruct` call on the master copy contract.
+
+It is **important** to know that it is possible to "hijack" a contract if the proxy creation and setup method are done in separate transactions. To avoid this it is possible to pass the initialisation data to the ProxyFactory or the DelegatingConstructorProxy.
+
+## Contracts
+
+### Base Contracts
+#### SelfAuthorized.sol
+The self authorized contract implements the `authorized()` so that only the contract itself is authorized to perform actions.
+
+Multiple contracts use the `authorized()` modifier. This modifier should be overwritten by contract to implemented the desired logic to check access to the protected methods.
+
+#### Proxy.sol
+The proxy contract implements only two functions: The constructor setting the address of the master copy and the fallback function forwarding all transactions sent to the proxy via a `DELEGATECALL` to the master copy and returning all data returned by the `DELEGATECALL`.
+
+#### DelegateConstructorProxy.sol
+This is an extension to the proxy contract that allows further initialization logic to be passed to the constructor.
+
+#### PayingProxy.sol
+This is an extension to the delegate constructor proxy contract that pays a specific amount to a target address after initialization.
+
+#### ProxyFactory.sol
+The proxy factory allows to create new proxy contracts pointing to a master copy and executing a function in the newly deployed proxy in one transaction. This additional transaction can for example execute the setup function to initialize the state of the contract.
+
+#### MasterCopy.sol
+The master copy contract defines the master copy field and has simple logic to change it. The master copy class should always be defined first if inherited.
+
+#### EtherPaymentFallback.sol
```

Comment:
I haven't found this contract in repo

Reply:
https://github.com/gnosis/safe-contracts/pull/42 will be in with this one